src/SSHDebugPS/Utilities/VSMessageBoxHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using ThreadingTasks = System.Threading.Tasks;
+
+namespace Microsoft.SSHDebugPS.Utilities
+{
+    internal class VSMessageBoxHelper
+    {
+        /// <summary>
+        /// Switches to main thread and shows a VS message box.
+        /// </summary>
+        public static void ShowMessage(string title, string message, OLEMSGICON icon, OLEMSGBUTTON button, OLEMSGDEFBUTTON defaultButton)
+        {
+            ThreadingTasks.Task.Run(async () =>
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
I can't tell if this is lambda is an `async Task` or `async void` method. I would probably suggest defining a local function instead to make it clear (`async void` is bad).